components/brave_extension/extension/brave_extension/background/events/dappDetectionEvents.ts

```diff
@@ -0,0 +1,19 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// content script listener for dapp detection.
+chrome.runtime.onMessage.addListener((msg, sender, sendResponse) => {
+  switch (msg.type) {
+    case 'dappAvailable': {
+      chrome.tabs.query({ active: true, currentWindow: true }, function (tabs: chrome.tabs.Tab[]) {
```

Comment:
`sender.url` has the url of the tab that sent the message

Reply:
also sender.tab.id contains the tab id so I think you want to compare the active tab id to the sender.tab.id

Reply:
When I implemented it, sender doesn't have `url`. Because of that I passed url by `msg.location`.

Reply:
it's missing because you have to use `onConnect` and then `port.onMessage.addListener` inside that

Reply:
updatd to not query anymore since msg.sender has the info we need. Also we don't need to check for active since the prompt is only tab level.  We had active before because we had a global dialog that popped up on muon.